src/main/scala/com.snowplowanalytics.snowplow.analytics.scalasdk/Event.scala

```diff
@@ -242,12 +242,202 @@ case class Event(
 
 object Event {
 
+  val FIELD_SIZES: Map[String, Int] = Map(
+    "app_id" -> 255,
+    "platform" -> 255,
+    "event" -> 128,
+    "event_id" -> 36,
+    "name_tracker" -> 128,
+    "v_tracker" -> 100,
+    "v_collector" -> 100,
+    "v_etl" -> 100,
+    "user_id" -> 255,
+    "user_ipaddress" -> 128,
+    "user_fingerprint" -> 128,
+    "domain_userid" -> 128,
+    "network_userid" -> 128,
+    "geo_country" -> 2,
+    "geo_region" -> 3,
+    "geo_city" -> 75,
+    "geo_zipcode" -> 15,
+    "geo_region_name" -> 100,
+    "ip_isp" -> 100,
+    "ip_organization" -> 128,
+    "ip_domain" -> 128,
+    "ip_netspeed" -> 100,
+    "page_url" -> 4096,
+    "page_title" -> 2000,
+    "page_referrer" -> 4096,
+    "page_urlscheme" -> 16,
+    "page_urlhost" -> 255,
+    "page_urlpath" -> 3000,
+    "page_urlquery" -> 6000,
+    "page_urlfragment" -> 3000,
+    "refr_urlscheme" -> 16,
+    "refr_urlhost" -> 255,
+    "refr_urlpath" -> 6000,
+    "refr_urlquery" -> 6000,
+    "refr_urlfragment" -> 3000,
+    "refr_medium" -> 25,
+    "refr_source" -> 50,
+    "refr_term" -> 255,
+    "mkt_medium" -> 255,
+    "mkt_source" -> 255,
+    "mkt_term" -> 255,
+    "mkt_content" -> 500,
+    "mkt_campaign" -> 255,
+    "se_category" -> 1000,
+    "se_action" -> 1000,
+    "se_label" -> 4096,
+    "se_property" -> 1000,
+    "tr_orderid" -> 255,
+    "tr_affiliation" -> 255,
+    "tr_city" -> 255,
+    "tr_state" -> 255,
+    "tr_country" -> 255,
+    "ti_orderid" -> 255,
+    "ti_sku" -> 255,
+    "ti_name" -> 255,
+    "ti_category" -> 255,
+    "useragent" -> 1000,
+    "br_name" -> 50,
+    "br_family" -> 50,
+    "br_version" -> 50,
+    "br_type" -> 50,
+    "br_renderengine" -> 50,
+    "br_lang" -> 255,
+    "br_colordepth" -> 12,
+    "os_name" -> 50,
+    "os_family" -> 50,
+    "os_manufacturer" -> 50,
+    "os_timezone" -> 255,
+    "dvce_type" -> 50,
+    "doc_charset" -> 128,
+    "tr_currency" -> 3,
+    "ti_currency" -> 3,
+    "base_currency" -> 3,
+    "geo_timezone" -> 64,
+    "mkt_clickid" -> 128,
+    "mkt_network" -> 64,
+    "etl_tags" -> 500,
+    "refr_domain_userid" -> 128,
+    "domain_sessionid" -> 128,
+    "event_vendor" -> 1000,
+    "event_name" -> 1000,
+    "event_format" -> 128,
+    "event_version" -> 128,
+    "event_fingerprint" -> 128
+  )
+
+  private def validateStr(
+    k: String,
+    value: String
+  ): List[String] =
+    if (value.length > FIELD_SIZES.getOrElse(k, Int.MaxValue))
+      List(s"Field $k value $value longer than maximum allowed size ${FIELD_SIZES.getOrElse(k, Int.MaxValue)}")
```

Comment:
The `$value` is a user-submitted string.  As a general rule, we want to phase out putting user data into stdout logs or into the free-form "msg" field of bad rows.  If you remove `$value` from the string then we are better able to use this message in logs or bad rows.